inbm-lib/inbm_common_lib/utility.py

```diff
@@ -180,11 +180,11 @@ def canonicalize_uri(url: str) -> CanonicalUri:
 
     WARNING: a string with no forward slashes will have a scheme added. e.g., 'a' -> 'https://a'
 
-    @param url: the url
+    @param url: URL
     @return canonicalized version"""
 
     if url and URL_NULL_CHAR in url:
-        raise UrlSecurityException("Unsafe characters detected in the url. Cannot process the request.")
+        raise UrlSecurityException("Unsafe characters detected in the URL. Cannot process the request.")
```

Comment:
**Security**
**Suggestion:** Replace the string concatenation with a more secure method to avoid potential security risks such as command injection.
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    if url and any(char in url for char in URL_NULL_CHAR):
        raise UrlSecurityException("Unsafe characters detected in the URL. Cannot process the request.")
```